nixos/modules/services/backup/mysql-backup.nix

```diff
@@ -5,12 +5,43 @@
   ...
 }:
 let
-
-  inherit (pkgs) mariadb gzip;
-
   cfg = config.services.mysqlBackup;
   defaultUser = "mysqlbackup";
 
+  compressionAlgs = {
+    gzip = rec {
+      pkg = pkgs.gzip;
+      ext = ".gz";
+      minLevel = 1;
+      maxLevel = 9;
+      cmd = compressionLevelFlag: "${pkg}/bin/gzip -c ${cfg.gzipOptions} ${compressionLevelFlag}";
+      validLevel = level: minLevel <= level && level <= maxLevel;
```

Comment:
`validLevel` would also make sense to not have duplicated for each algorithm, but it's fine to leave it like this too